src/descriptive.js

```diff
@@ -28,7 +28,23 @@ export function max(arr) {
   return result;
 }
 
+/**
+ * Returns the product of all entries in a numeric data array.
+ * @param {Array} arr the data array
+ * @returns {Number} the product of all the data in the array
+ */
+export function product(arr) {
+  if (!Array.isArray(arr) || arr.length === 0) return undefined;
+  let result = 1;
+  for (let i = 0; i < arr.length; i += 1) {
+    if (!Number.isFinite(arr[i])) return undefined;
+    result = result * arr[i];
```

Comment:
Please fix code patterns issue as highlighted by codacy.
```suggestion
    result *= arr[i];
```